kotlinx-coroutines-core/common/test/sync/MutexTest.kt

```diff
@@ -5,8 +5,10 @@
 package kotlinx.coroutines.sync
 
 import kotlinx.coroutines.*
+import kotlin.contracts.ExperimentalContracts
 import kotlin.test.*
 
+@ExperimentalContracts
```

Comment:
Why is it needed in tests and examples?  Something is wrong here.

Reply:
The `contract` is marked with the annotation `@ExperimentalContracts`.     
@ExperimentalContracts have propagatinging, so tests and examples also need to be marked @ExperimentalContracts. (See [Opt-in Requirements](https://kotlinlang.org/docs/reference/opt-in-requirements.htm)l) 
My idea is that `Mutex.withLock()` / `Semaphore.withPermit()` is an open API, and users have the right to decide whether to use this Experimental funcation, rather than the library developers directly blocking its propagate.

Reply:
That's a no-go. Both `withLock` and `withPermit` are public. They cannot force their users to opt-in. However, both `withLock` and `withPermit` can opt-in into experimental contracts without forcing users to.

Reply:
Updated. [752e060](https://github.com/Kotlin/kotlinx.coroutines/pull/2010/commits/752e0604d882eec219831dfe91c3cf2db0b3535b)